### PR TITLE
Rework API key UI and API docs for the election API

### PR DESCRIPTION
**Pull request by benweissmann**
- Restrict access to the Action API (previously, we didn't need to do this because we only allowed subscribers with Action API access to create API keys)
- Open up API key management UI to all subscribers with either election API access (any subscriber not on the free plan) OR the action API
- Update the docs to reflect the new authenticated API endpoints

### CODE REVIEW

File: app/apikey/manage_views.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 class ApiAccessViewMixin(SubscriberManageViewMixin):
     def check_subscriber_access(self):
-        if not self.subscriber.has_api_access:
+        if not self.subscriber.has_api_access():
             raise Http404
```
Comment (ericisaiah) [inline app/apikey/manage_views.py:27]: I know this was not a change you made, but might be good to make these Http403 moving forward.

**Review by ericisaiah — COMMENTED** [review]
Thanks for walking through it with me. Besides the minor comment about http codes, and addressing the CK issue, my only other question is about the OptOutView and OptInView classes. Should those also be protected by the action API authorization check?

**Comment by benweissmann** [comment]
The OptInView and OptOutView are endpoint that we use internally; they have a check at the top for `request.auth.subscriber.is_first_party` (which indicates that the API key belongs to VoteAmerica, and not a subscriber) -- but we could refactor that check to be its own `FirstPartyApiKeyRequired` permission class.

**Comment by ericisaiah** [comment]
Good idea - I'd vote for that to be consistent with auth across API endpoints.

**Comment by benweissmann** [comment]
👍 updated

**Review by ericisaiah — COMMENTED** [review]
Looks good

**Pull request merged**